be/src/olap/rowset/segment_v2/segment_iterator.cpp

```diff
@@ -45,16 +47,21 @@ SegmentIterator::~SegmentIterator() {
 }
 
 Status SegmentIterator::init(const StorageReadOptions& opts) {
+    DorisMetrics::segment_read_total.increment(1);
     _opts = opts;
     RETURN_IF_ERROR(_init_short_key_range());
+    RETURN_IF_ERROR(_init_row_ranges());
     RETURN_IF_ERROR(_init_column_iterators());
     return Status::OK();
 }
 
 // This function will use input key bounds to get a row range.
 Status SegmentIterator::_init_short_key_range() {
+    DorisMetrics::segment_row_total.increment(num_rows());
     _lower_rowid = 0;
     _upper_rowid = num_rows();
+    // initial short key row ranges: [0, num_rows())
+    _row_ranges = RowRanges::create_single(_lower_rowid, _upper_rowid);
```

Comment:
do we need this?

Reply:
yes, when _opts.lower_bound == nullptr && _opts.upper_bound == nullptr is true, we should read the whole segment with row ranges [0, num_rows)